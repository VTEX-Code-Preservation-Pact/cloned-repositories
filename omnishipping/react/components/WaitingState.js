import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Skeleton from 'react-loading-skeleton'
import { injectIntl, intlShape } from 'react-intl'
import { translate } from './../utils/i18nUtils'

export class WaitingState extends Component {
  render() {
    const { intl, loading } = this.props

    return (
      <div className="step accordion-group shipping-data">
        <div className="accordion-heading">
          <span className="accordion-toggle collapsed">
            <i className="icon-home" /> {translate(intl, 'deliveryTitle')}
          </span>
        </div>
        <div className="accordion-inner">
          <div className="box-step box-info shipping-summary-placeholder">
            <div className="shipping-summary-info">
              {loading ? <Skeleton /> : translate(intl, 'awatingData')}
            </div>
          </div>
        </div>
      </div>
    )
  }
}

WaitingState.defaultProps = {
  loading: false,
}

WaitingState.propTypes = {
  intl: intlShape,
  loading: PropTypes.bool,
}

export default injectIntl(WaitingState)
