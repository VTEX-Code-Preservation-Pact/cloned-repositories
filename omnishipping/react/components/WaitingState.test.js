import React from 'react'
import { shallowWithIntl, loadTranslation } from 'enzyme-react-intl'

import { WaitingState } from './WaitingState'

loadTranslation('../../messages/en.json')

describe('WaitingState', () => {
  it('should render self and subcomponents', () => {
    const wrapper = shallowWithIntl(<WaitingState />)

    expect(wrapper.getElement()).toMatchSnapshot()
  })
})
