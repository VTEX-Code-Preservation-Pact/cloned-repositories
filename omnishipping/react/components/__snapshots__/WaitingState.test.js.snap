// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaitingState should render self and subcomponents 1`] = `
<div
  className="step accordion-group shipping-data"
>
  <div
    className="accordion-heading"
  >
    <span
      className="accordion-toggle collapsed"
    >
      <i
        className="icon-home"
      />
       
      Shipping
    </span>
  </div>
  <div
    className="accordion-inner"
  >
    <div
      className="box-step box-info shipping-summary-placeholder"
    >
      <div
        className="shipping-summary-info"
      >
        Waiting for more information
      </div>
    </div>
  </div>
</div>
`;
